### PR TITLE
Acknowledge Incident In PagerDuty

### PR DESCRIPTION
**Pull request by mk-armah**
Added github action to enable acknowlegment of pagerduty incident from port
This action takes `from` as input parameter and leverages incident details present in port to update pagerduty incident status to acknowleged

## Inputs
| Name                 | Description                                                                                          | Required | Default            |
|----------------------|------------------------------------------------------------------------------------------------------|----------|--------------------
| from              | The email address of a valid user associated with the account making the request.                                                              | true    | -               |

### CODE REVIEW

File: github-workflows/pagerduty/README.md [hunk 1]
```diff
@@ -477,4 +477,352 @@ jobs:
...
+
+      - name: Log After Upserting Entity
+        uses: port-labs/port-github-action@v1
+        with:
+          clientId: ${{ secrets.PORT_CLIENT_ID }}
+          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
+          baseUrl: https://api.getport.io
+          operation: PATCH_RUN
+          runId: ${{fromJson(github.event.inputs.port_payload).context.runId}}
+          logMessage: "Entity upserting was successful ✅"
```
Comment (yairsimantov20) [inline github-workflows/pagerduty/README.md:825]: Can you also make sure to send a log for a failure?

File: github-workflows/pagerduty/README.md [hunk 2]
```diff
@@ -477,4 +477,352 @@ jobs:
...
+
+      - name: Inform completion of pagerduty feature flag ingestion into Port
+        uses: port-labs/port-github-action@v1
+        with:
+          clientId: ${{ secrets.PORT_CLIENT_ID }}
+          clientSecret: ${{ secrets.PORT_CLIENT_SECRET }}
+          baseUrl: https://api.getport.io
+          operation: PATCH_RUN
+          runId: ${{fromJson(github.event.inputs.port_payload).context.runId}}
+          logMessage: "Entity upserting was successful ✅"
```
Comment (yairsimantov20) [inline github-workflows/pagerduty/README.md:665]: Can you also make sure to send a log for a failure?

File: github-workflows/pagerduty/README.md [hunk 3]
```diff
@@ -477,4 +477,352 @@ jobs:
...
+        description: >-
+          Port's payload, including details for who triggered the action and
+          general context (blueprint, run id, etc...)
+
+jobs:
+  change-incident-owner:
+    runs-on: ubuntu-latest
+    steps:
+      
+      - name: Inform execution of request to change incident owner
```
Comment (yairsimantov20) [inline github-workflows/pagerduty/README.md:593]:
```suggestion
    steps:
      - name: Inform execution of request to change incident owner
```

File: github-workflows/pagerduty/README.md [hunk 4]
```diff
@@ -477,4 +477,352 @@ jobs:
...
+          Port's payload, including details for who triggered the action and
+          general context (blueprint, run id, etc...)
+
+jobs:
+  acknowledge_incident:
+    runs-on: ubuntu-latest
+    steps:
+      
+      - name: Log Executing Request to Acknowledge Incident
+        uses: port-labs/port-github-action@v1
```
Comment (yairsimantov20) [inline github-workflows/pagerduty/README.md:748]:
```suggestion
    steps:
      - name: Log Executing Request to Acknowledge Incident
        uses: port-labs/port-github-action@v1
```

File: github-workflows/pagerduty/README.md [hunk 5]
```diff
@@ -477,4 +477,352 @@ jobs:
...
+        type: string
+      from:
+        description: The email address of a valid user associated with the account making the request.
+        required: true
+        type: string
+      port_payload:
+        required: true
+        description: >-
+          Port's payload, including details for who triggered the action and
+          general context (blueprint, run id, etc...)
```
Comment (yairsimantov20) [inline github-workflows/pagerduty/README.md:586]: Inputs are set to required but you are not validating it
github wont validate it for you

File: github-workflows/pagerduty/README.md [hunk 6]
```diff
@@ -477,4 +477,352 @@ jobs:
...
+    inputs:
+      from:
+        description: The email address of a valid user associated with the account making the request.
+        required: true
+        type: string
+      port_payload:
+        required: true
+        description: >-
+          Port's payload, including details for who triggered the action and
+          general context (blueprint, run id, etc...)
```
Comment (yairsimantov20) [inline github-workflows/pagerduty/README.md:740]: Inputs are set to required but you are not validating it
github wont validate it for you

**Review by yairsimantov20 — CHANGES_REQUESTED** [review]
Left comments
Also please resolve conflicts

File: github-workflows/pagerduty/README.md (hunk 6 above)
Comment (mk-armah) [reply github-workflows/pagerduty/README.md:739]: How about Port's validation ?

**Pull request closed without merging**